torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -315,7 +315,14 @@ void RequestCallbackImpl::processRpc(
                          .type();
       }
 
-      auto ownerRRef = ctx.getOrCreateOwnerRRef(rrefId, returnType);
+      c10::intrusive_ptr<OwnerRRef> ownerRRef;
+      if (rrefId == forkId) {
+        // Creating an owner RRef on self, should already exist in owners map
+        ownerRRef =
+            ctx.getOwnerRRef(rrefId, /* forceCreated */ true)->constValue();
```

Comment:
Just confirming, will the exception thrown by `TORCH_INTERNAL_ASSERT` in `getOwnerRRef` be captured by the try-catch block in `RequestCallbackImpl::processMessage`, so that we don't lose a thread?

Reply:
Yep, this error is indeed caught and processed by `RequestCallback::handleError` which logs the error and sets it on the future, and the thread doesn't crash. We do see the error and stacktrace logged such as:

```
> E0715 13:38:22.403939 1125458 request_callback_impl.cpp:168] Received error while processing request type 5: false INTERNAL ASSERT FAILED at "ca
ffe2/torch/csrc/distributed/rpc/rref_context.cpp":379, please report a bug to PyTorch. Expected OwnerRRef with id GloballyUniqueId(created_on=0, l
ocal_id=0) to be created.
```